### ISSUE TITLE
In the Properties panel, the "remove keyframe" option is shown in the context menu even when it is not applicable

### ISSUE THREAD

**Issue by raindropsfromsky** [OpenShot/openshot-qt#5438]
**Describe the bug:**
In the Properties panel, the "remove keyframe" option is shown in the context menu even when it is not applicable, because there is no keyframe to be removed.

**Steps to reproduce the behavior:**
1. In the Properties panel, select any parameter where keyframe can be inserted (e.g. Alpha)
2. Right click. The options include "insert keyframe" and "remove keyframe".
    But we have not inserted any keyframes yet. So why is this options shown?

**Expected behavior:**
The "remove keyframe" option should be shown only when a keyframe is inserted previously.
(state-dependent behavior)

**System Details:**
- OpenShot-v3.1.1- daily- 11796- d98b5f2f-8e9d7edc Build Date: 2024-02-19 22:14
- Windows 11

**Comment by Colorjet3** [comment]
There are many other attributes (Parent for instance) that also show up in the Properties window where it does nothing.

This, again, will be implemented as part of the new UI redesign phase.

**Comment by raindropsfromsky** [comment]
The new design can have two different approaches:
1. Gray out the non-appliable options 
2. Remove the non-applicable options

In the first approach, all options in the context menu remain in their original positions.
That allows us to develop muscle memory.

If non-applicable options are removed, the context menu will keep changing.
Then the user has to look for his desired option carefully.

Just my 2 cents!

**Comment by Colorjet3** [comment]
Yeah, I can argue against just Graying out.  The reason being is that now I am not sure if this attribute is just locked because it is not required or is it locked but the values do change and they mean something to me.

We will just have to see what the lead developer decides.

Thanks again for the input.

**Comment by raindropsfromsky** [comment]
Just to clarify, if the option is not applicable, it should not appear at all.
I am not saying that the menu should show all options at all times!

But show a grayed out option if it is applicable but locked.

**Comment by Colorjet3** [comment]
Actually, you will see this soon in one of the latest DEV Daily builds.  Hopefully in the next 2-3 weeks.

**Update:**  Not everywhere, but I will point out to you so you can provide feedback.
